datasource_api_client/api/datasource/post_datasource.py

```diff
@@ -0,0 +1,108 @@
+from typing import Any, Dict, Optional
+
+import httpx
+
+from ...client import Client
+from ...models.create_datasource_request import CreateDatasourceRequest
+from ...models.datasource_dto import DatasourceDto
+from ...types import Response
+
+
+def _get_kwargs(
+    *,
+    client: Client,
+    json_body: CreateDatasourceRequest,
+) -> Dict[str, Any]:
+    url = "{}/".format(client.base_url)
+
+    headers: Dict[str, Any] = client.get_headers()
+    cookies: Dict[str, Any] = client.get_cookies()
+
+    json_json_body = json_body.to_dict()
+
+    return {
+        "url": url,
+        "headers": headers,
+        "cookies": cookies,
+        "timeout": client.get_timeout(),
+        "json": json_json_body,
+        "verify": client.verify_ssl,
+    }
+
+
+def _parse_response(*, response: httpx.Response) -> Optional[DatasourceDto]:
+    if response.status_code == 200:
+        response_200 = DatasourceDto.from_dict(response.json())
+
+        return response_200
+    return None
+
+
+def _build_response(*, response: httpx.Response) -> Response[DatasourceDto]:
+    return Response(
+        status_code=response.status_code,
+        content=response.content,
+        headers=response.headers,
+        parsed=_parse_response(response=response),
+    )
+
+
+def sync_detailed(
+    *,
+    client: Client,
+    json_body: CreateDatasourceRequest,
+) -> Response[DatasourceDto]:
+    kwargs = _get_kwargs(
+        client=client,
+        json_body=json_body,
+    )
+
+    response = httpx.post(
+        **kwargs,
+    )
+
+    return _build_response(response=response)
+
+
+def sync(
+    *,
+    client: Client,
+    json_body: CreateDatasourceRequest,
+) -> Optional[DatasourceDto]:
+    """ """
+
+    return sync_detailed(
+        client=client,
+        json_body=json_body,
+    ).parsed
+
+
+async def asyncio_detailed(
+    *,
+    client: Client,
+    json_body: CreateDatasourceRequest,
+) -> Response[DatasourceDto]:
+    kwargs = _get_kwargs(
+        client=client,
+        json_body=json_body,
+    )
+
+    async with httpx.AsyncClient() as _client:
+        response = await _client.post(**kwargs)
+
+    return _build_response(response=response)
+
+
+async def asyncio(
+    *,
+    client: Client,
+    json_body: CreateDatasourceRequest,
+) -> Optional[DatasourceDto]:
+    """ """
+
+    return (
+        await asyncio_detailed(
+            client=client,
+            json_body=json_body,
+        )
+    ).parsed
```

Comment:
would love to hear more about this in a what I learnt this sprint session - I think I get the gist of it but it's interesting!